### PR TITLE
Check argument size on static array parameters

### PR DESCRIPTION
**Pull request by wrongnull**
Closes https://github.com/Vexu/arocc/issues/349

### CODE REVIEW

File: src/aro/Parser.zig [hunk 1]
```diff
@@ -7683,6 +7684,18 @@ fn callExpr(p: *Parser, lhs: Result) Error!Result {
             continue;
         }
         const p_ty = params[arg_count].ty;
+        if (p_ty.statically_sized_array) {
+            if (arg.ty.data == .array and arg.ty.data.array.*.len < p_ty.data.array.*.len) {
```
Comment (Vexu) [inline src/aro/Parser.zig:7688]: This should use `arrayLen` instead of making the union tagged.

File: src/aro/Parser.zig [hunk 2]
```diff
@@ -7683,6 +7684,18 @@ fn callExpr(p: *Parser, lhs: Result) Error!Result {
...
+                const extra = Diagnostics.Message.Extra{ .arguments = .{
+                    .expected = @intCast(arg.ty.data.array.*.len),
+                    .actual = @intCast(p_ty.data.array.*.len),
+                } };
+                try p.errExtra(.array_argument_too_small, param_tok, extra);
```
Comment (Vexu) [inline src/aro/Parser.zig:7693]: Clang also adds a `callee declares array parameter as static here` note. You should be able to do the same with `params[arg_count].name_tok`.

File: src/aro/Type.zig [hunk 3]
```diff
@@ -425,6 +425,7 @@ data: union {
 specifier: Specifier,
 qual: Qualifiers = .{},
 decayed: bool = false,
+statically_sized_array: bool = false,
```
Comment (Vexu) [inline src/aro/Type.zig:428]: `static_array` specifier exists but is seemingly unused currently. Could you try using that instead?

(Apparently it was only used for three days between c35d5c22f7adc314202c5dcd220d88d6ff7d068d and 9b4daaa8647d8ea7c24ed6edae66a32433fb4ba6)

**Review by Vexu — CHANGES_REQUESTED** [review]

**Comment by wrongnull** [comment]
@Vexu all the desired changes have been made

**Pull request merged**